extensions/interactions/DragAndDropSortInput/DragAndDropSortInput.py

```diff
@@ -79,3 +81,203 @@ class DragAndDropSortInput(base.BaseInteraction):
+                interaction.customization_args)
+        )
+
+        outcome_proto = interaction.default_outcome.to_proto()
+
+        hints_proto_list = cls.get_hint_proto(cls, interaction.hints)
+
+        solution_proto = cls._to_solution_proto(
+            interaction.solution)
+
+        answer_groups_proto = (
+            cls._to_answer_groups_proto(
+                interaction.answer_groups)
+        )
+
+        drag_and_drop_interaction_proto = (
+            state_pb2.DragAndDropSortInputInstance(
+                customization_args=customization_args_proto,
+                hints=hints_proto_list,
+                default_outcome=outcome_proto,
+                solution=solution_proto,
+                answer_groups=answer_groups_proto
+            )
+        )
+
+        return drag_and_drop_interaction_proto
+
+    @classmethod
+    def _to_answer_groups_proto(cls, answer_groups):
+        """Creates a AnswerGroup proto object
+            for DragAndDropSortInputInstance.
+
+        Args:
+            answer_groups: list(AnswerGroup). List of answer groups of the
+                interaction instance.
+
+        Returns:
+            list. The AnswerGroup proto object list.
+        """
+        answer_group_list_proto = []
+
+        for answer_group in answer_groups:
+            base_answer_group_proto = answer_group.to_proto()
+            answer_group_proto = (
+                state_pb2.DragAndDropSortInputInstance.AnswerGroup(
+                    base_answer_group=base_answer_group_proto
+                )
+            )
+            answer_group_list_proto.append(answer_group_proto)
+
+        return answer_group_list_proto
+
+    @classmethod
+    def _to_solution_proto(cls, solution):
+        """Creates a Solution proto object
+            for DragAndDropSortInputInstance.
+
+        Args:
+            solution: Solution. A possible solution
+                for the question asked in this interaction.
+
+        Returns:
+            Solution. The Solution proto object.
+        """
+        solution_proto = None
+        if solution is not None:
+            solution_proto = (
+                state_pb2.DragAndDropSortInputInstance.Solution(
+                    base_solution=solution.to_proto(),
+                    correct_answer=(
+                        cls._to_list_of_set_of_translatable_html_content_ids(
+                            solution.correct_answer)
+                    )
+                )
+            )
+
+        return solution_proto
+
+    @classmethod
+    def _to_list_of_set_of_translatable_html_content_ids(
+        cls, correct_answer
+    ):
+        """Creates a ListOfSetsOfTranslatableHtmlContentIds proto object.
+
+        Args:
+            correct_answer: list. A list of set of
+                TranslatableHtmlContentId.
+
+        Returns:
+            ListOfSetsOfTranslatableHtmlContentIds. The
+            ListOfSetsOfTranslatableHtmlContentIds proto object.
```

Comment:
I don't think it is necessary to repeat the type info, lets just say `The proto object.`

Reply:
ditto elsewhere

Reply:
Fixed